File: app/src/main/java/com/cdat/pre_login/PreLoginDashbordFragment.java
package com.cdat.pre_login;

import android.animation.Animator;
import android.animation.AnimatorListenerAdapter;
import android.animation.AnimatorSet;
import android.animation.ObjectAnimator;
import android.os.Bundle;
import android.os.SystemClock;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import com.cdat.R;
import com.cdat.fragments.LoginFragment;
import com.cdat.fragments.SignUpFragment;
import com.cdat.helper.Config;


public class PreLoginDashbordFragment extends Fragment implements View.OnClickListener {

    private LinearLayout llLogin, llContactUs, llTechSupport, llUsrManual;
    private long mLastClickTime = 0;

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        try {
            //  ((AppCompatActivity) getActivity()).getSupportActionBar().setTitle(m_config.lbl_heading);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_pre_login_dashboard, container, false);

        try {
            llLogin = (LinearLayout) rootView.findViewById(R.id.ll_login);
            llContactUs = (LinearLayout) rootView.findViewById(R.id.ll_contactUs);
            llTechSupport = (LinearLayout) rootView.findViewById(R.id.ll_techSupport);
            llUsrManual = (LinearLayout) rootView.findViewById(R.id.ll_usrManual);

            final AnimatorSet mAnimationSet = new AnimatorSet();

            ObjectAnimator moveY_tl_otp = ObjectAnimator.ofFloat(llLogin, "translationX", -1500f, 0);
            moveY_tl_otp.setDuration(1500);

            ObjectAnimator moveY_tl_otp2 = ObjectAnimator.ofFloat(llContactUs, "translationX", 1500f, 0);
            moveY_tl_otp2.setDuration(1500);

            ObjectAnimator moveY_tl_otp3 = ObjectAnimator.ofFloat(llTechSupport, "translationX", -1500f, 0);
            moveY_tl_otp3.setDuration(1500);

            ObjectAnimator moveY_tl_otp4 = ObjectAnimator.ofFloat(llUsrManual, "translationX", 1500f, 0);
            moveY_tl_otp4.setDuration(1500);

            mAnimationSet.play(moveY_tl_otp);
            mAnimationSet.play(moveY_tl_otp2);
            mAnimationSet.play(moveY_tl_otp3);
            mAnimationSet.play(moveY_tl_otp4);
            mAnimationSet.addListener(new AnimatorListenerAdapter() {
                @Override
                public void onAnimationEnd(Animator animation) {
                    super.onAnimationEnd(animation);

                }
            });
            mAnimationSet.start();
            llLogin.setOnClickListener(this);
            llContactUs.setOnClickListener(this);
            llTechSupport.setOnClickListener(this);
            llUsrManual.setOnClickListener(this);

        } catch (Exception e) {

        }

        llLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
//                SignUpFragment registerFragment = new SignUpFragment();
//                Config.slideFragment(registerFragment, "", getFragmentManager(), R.id.content_frame, "ankole")
            }
        });
        return rootView;
    }

    @Override
    public void onClick(View view) {

        if (SystemClock.elapsedRealtime() - mLastClickTime < 1000) {
            return;
        }
        mLastClickTime = SystemClock.elapsedRealtime();
        try {
            switch (view.getId()) {

                case R.id.ll_login:

//                    BmiCalculatorFragment bmiCalFrag = new BmiCalculatorFragment();
//                    Config.slideFragment(bmiCalFrag, "", getFragmentManager(), R.id.fragment_placeholder, "2");
                    break;
                case R.id.ll_contactUs:
//                    BMRCalculatorFragment bmrCalculatorFragment = new BMRCalculatorFragment();
//                    Config.slideFragment(bmrCalculatorFragment, "", getFragmentManager(), R.id.fragment_placeholder, "2");
                    break;
                case R.id.ll_techSupport:
//                    PregnancyCalculatorFragment pregFrag = new PregnancyCalculatorFragment();
//                    Config.slideFragment(pregFrag, "", getFragmentManager(), R.id.fragment_placeholder, "2");
                    break;
                case R.id.ll_usrManual:
//                    FertilityCalculatorFragment fertilityCalFragment = new FertilityCalculatorFragment();
//                    Config.slideFragment(fertilityCalFragment, "", getFragmentManager(), R.id.fragment_placeholder, "2");
                    break;

                default:
                    break;

            }
        } catch (Exception e) {

        }
    }
}

File: app/src/main/java/com/cdat/pre_login/PreLoginDashbordFragment.kt
package com.cdat.pre_login

import android.content.Intent
import android.os.Bundle
import android.os.SystemClock
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.animation.AnimationUtils
import android.widget.LinearLayout
import com.cdat.R
import com.cdat.activity.HomeActivity
import com.cdat.contact_us.ContactUsFragment
import com.cdat.contact_us.TechnicalSupportFragment
import com.cdat.fragments.HomeFragment
import com.cdat.fragments.LoginFragment
import com.cdat.fragments.SignUpFragment
import com.cdat.helper.Config


class PreLoginDashbordFragment : Fragment() {

    private lateinit var llLogin: LinearLayout
    private lateinit var llRegistration: LinearLayout
    private lateinit var llContactUs: LinearLayout
    private lateinit var llTechSupport: LinearLayout
    private lateinit var llUsrManual: LinearLayout
    private var mLastClickTime: Long = 0

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        try {
            //  ((AppCompatActivity) getActivity()).getSupportActionBar().setTitle(m_config.lbl_heading);
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val rootView = inflater.inflate(R.layout.fragment_pre_login_dashboard, container, false)

        try {
            llLogin = rootView.findViewById<LinearLayout>(R.id.ll_login) as LinearLayout
            llRegistration = rootView.findViewById<LinearLayout>(R.id.ll_registration) as LinearLayout
            llContactUs = rootView.findViewById<LinearLayout>(R.id.ll_contactUs) as LinearLayout
            llTechSupport = rootView.findViewById<LinearLayout>(R.id.ll_techSupport) as LinearLayout
            llUsrManual = rootView.findViewById<LinearLayout>(R.id.ll_usrManual) as LinearLayout


            llLogin!!.setAnimation(AnimationUtils.loadAnimation(activity, R.anim.slide_in_up))
            llRegistration!!.setAnimation(AnimationUtils.loadAnimation(activity, R.anim.slide_in_down))
            llContactUs!!.setAnimation(AnimationUtils.loadAnimation(activity, R.anim.slide_in_up))
            llTechSupport!!.setAnimation(AnimationUtils.loadAnimation(activity, R.anim.slide_out_left))
            llUsrManual!!.setAnimation(AnimationUtils.loadAnimation(activity, R.anim.slide_in_up))



        } catch (e: Exception) {

        }

        llLogin!!.setOnClickListener {
            val loginFragment = LoginFragment()
            Config.slideFragment(loginFragment, "", fragmentManager!!, R.id.content_frame, "2")
        }
        llRegistration!!.setOnClickListener {
            val registrationFragment = SignUpFragment()
            Config.slideFragment(registrationFragment, "", fragmentManager!!, R.id.content_frame, "2")
        }
        llContactUs!!.setOnClickListener {
//            Toast.makeText(getActivity(), "Coming soon", Toast.LENGTH_SHORT).show();
            val contactusFragment = ContactUsFragment()
            Config.slideFragment(contactusFragment, "", fragmentManager!!, R.id.content_frame, "2")
        }
        llTechSupport!!.setOnClickListener {
//            Toast.makeText(getActivity(), "Coming soon", Toast.LENGTH_SHORT).show();
            val technicalsupFragment = TechnicalSupportFragment()
            Config.slideFragment(technicalsupFragment, "", fragmentManager!!, R.id.content_frame, "2")
        }
        llUsrManual!!.setOnClickListener {
            startActivity(Intent(activity, HomeActivity::class.java))
            activity!!.finish()
            activity!!.overridePendingTransition(R.anim.slide_in_right, R.anim.slide_out_left)
        }
        return rootView
    }

}
